pkg/api/pod/util.go

```diff
@@ -1584,6 +1589,13 @@ func useOnlyRecursiveSELinuxChangePolicy(oldPodSpec *api.PodSpec) bool {
 	return true
 }
 
+func hasUserNamespacesWithVolumeDevices(podSpec *api.PodSpec) bool {
+	if errs := apivalidation.ValidateHostUsersVolumeDevices(podSpec, nil); len(errs) > 0 {
```

Comment:
actually narrowly check if the pod sets host users to false and has volume devices rather than calling a potentially broad validation method and assuming any errors are related to this specific thing

```go
if podSpec.HostUsers == nil || podSpec.HostUsers == true {
  return false
}
hasVolumeDevices := false
pod.VisitContainers(podSpec, AllContainers, func(c *v1.Container, _ ContainerType) bool {
  if len(c.VolumeDevices) > 0 {
    hasVolumeDevices = true
    return false // stop iterating
  }
  return true // keep iterating
})
return hasVolumeDevices
```

Reply:
Thanks, fixed!